packages/test-generator/integration-test-templates/cypress/e2e/form/DSAllSupportedFormFields-spec.cy.ts

```diff
@@ -242,4 +242,122 @@ describe('FormTests - DSAllSupportedFormFields', () => {
+      awsIPAddressField.should('have.value', '123.12.34.56');
+      awsIPAddressField.type('{backspace}{backspace}78');
+      awsIPAddressField.should('have.value', '123.12.34.78');
+
+      // Boolean
+      cy.contains('Boolean').children('[data-checked="true"]').should('exist');
+      cy.contains('Boolean').click();
+      cy.contains('Boolean').children('[data-checked="false"]').should('exist');
+
+      const awsJsonField = getTextAreaByLabel('Aws json');
+      awsJsonField.should('have.value', JSON.stringify({ myKey: 'myValue' }));
+      awsJsonField.type('{backspace},"secondKey":"secondValue"}');
+      awsJsonField.should('have.value', JSON.stringify({ myKey: 'myValue', secondKey: 'secondValue' }));
+
+      const awsPhoneField = getInputByLabel('Aws phone');
+      awsPhoneField.should('have.value', '713 343 5938');
+      awsPhoneField.type('{backspace}{backspace}{backspace}{backspace}5678');
+      awsPhoneField.should('have.value', '713 343 5678');
+
+      // Enum
+      cy.get('select').should('have.value', 'NEW_YORK');
+      cy.get('select').select('Austin');
+      cy.get('select').should('have.value', 'AUSTIN');
+
+      const nonModelField = getTextAreaByLabel('Non model field');
+      nonModelField.should('have.value', JSON.stringify({ StringVal: 'myValue' }));
+      nonModelField.type('{backspace},"secondKey":"secondValue"}');
+      nonModelField.should('have.value', JSON.stringify({ StringVal: 'myValue', secondKey: 'secondValue' }));
+
+      getArrayFieldButtonByLabel('Non model field array').click();
+      getTextAreaByLabel('Non model field array').type(JSON.stringify({ StringVal: 'index1StringValue' }), {
+        parseSpecialCharSequences: false,
+      });
+      clickAddToArray();
+
+      cy.contains('Submit').click();
+
+      cy.contains(/Update1String/).then((recordElement: JQuery) => {
+        const record = JSON.parse(recordElement.text());
+
+        expect(record.string).to.equal('Update1StringX');
+        expect(record.stringArray).to.deep.equal(['String1', 'String2']);
+        expect(record.int).to.equal(10123);
+        expect(record.float).to.equal(4.3456);
+        expect(record.awsDate).to.equal('2023-02-13');
+        expect(record.awsTime).to.equal('12:34:56.789');
+        expect(record.awsDateTime).to.equal('2023-01-13T11:11:00.000Z');
+        expect(record.awsTimestamp).to.equal(1000000001);
+        expect(record.awsEmail).to.equal('myemail@amazon.org');
+        expect(record.awsUrl).to.equal('https://www.google.com');
+        expect(record.awsIPAddress).to.equal('123.12.34.78');
+        expect(record.boolean).to.equal(false);
+        expect(record.awsJson).to.deep.equal({ myKey: 'myValue', secondKey: 'secondValue' });
+        expect(record.awsPhone).to.equal('713 343 5678');
+        expect(record.enum).to.equal('AUSTIN');
+        expect(record.nonModelField).to.deep.equal({ StringVal: 'myValue', secondKey: 'secondValue' });
```

Comment:
This is sadly a limitation of our testing environment that we can't account for validation from AppSync. 
If you were to add "secondKey" to nonModelField, it would fail at AppSync.
You can update this to be more faithful to that scenario by conforming to the CustomType shape.